compiler/rustc_mir/src/transform/inline.rs

```diff
@@ -42,7 +42,7 @@ impl<'tcx> MirPass<'tcx> for Inline {
         // If you change this optimization level, also change the level in
         // `mir_drops_elaborated_and_const_checked` for the call to `mir_inliner_callees`.
         // Otherwise you will get an ICE about stolen MIR.
-        if tcx.sess.opts.debugging_opts.mir_opt_level < 2 {
+        if tcx.sess.opts.debugging_opts.mir_opt_level == 0 {
```

Comment:
could you rerun this experiment, but only enable the optimization in release mode?